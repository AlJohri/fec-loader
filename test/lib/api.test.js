const { ObjectWritableMock } = require('stream-mock');

jest.mock('isomorphic-unfetch');
const fetch = require('isomorphic-unfetch');

const api = require('../../lib/api');

test('api', async done => {
    fetch.mockReturnValue(
        Promise.resolve({
            json: () =>
                Promise.resolve({
                    results: [
                        {
                            fec_file_id: 'FEC-24'
                        }
                    ]
                })
        })
    );

    const writer = new ObjectWritableMock();

    const stream = await api({
        key: 'example'
    });

    stream.pipe(writer);

    writer.on('finish', () => {
        expect(writer.data).toMatchSnapshot();
        done();
    });
    
});
