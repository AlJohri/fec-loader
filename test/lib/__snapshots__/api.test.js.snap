// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api 1`] = `
Array [
  Object {
    "fec_file_id": 24,
  },
]
`;
